crates/runtime/src/lib.rs

```diff
@@ -210,8 +214,15 @@ impl Runtime {
                         continue;
                     }
                 };
-
                 tracing::info!("Loaded dataset: {}", &ds.name);
+                let engine_str = ds.acceleration.clone().map(|acc| {
+                    if acc.enabled {
+                        acc.engine().clone().to_string()
+                    } else {
+                        "None".to_string()
+                    }
+                });
+                metrics::gauge!("datasets/count", "engine" => engine_str.unwrap_or("None".to_string())).increment(1.0);
```

Comment:
Does this work?
```suggestion
                let engine_str = ds.acceleration.as_ref().map_or_else(
                    || "None", // Default value for the None case
                    |acc| if acc.enabled {
                        acc.engine().to_string().as_str() // Transform if Some and enabled
                    } else {
                        "None" // Return "None" if not enabled
                    }
                );
                metrics::gauge!("datasets/count", "engine" => engine_str.unwrap_or("None".to_string())).increment(1.0);
```

Reply:
I had issue with borrowing temporary value.